### PR TITLE
81205: Add encryption state log for Decision Review evidence files

### PR DESCRIPTION
**Pull request by dfong-adh**
## Summary

To help with analyzing downstream errors related to Decision Reviews evidence file uploads, we are updating the logs to include the encryption status of evidence files before they are stored in S3. 

No feature toggle is used for this as this is adding the presence of a boolean to logs.

I work for the Benefits Decision Reviews team.

## Related issue(s)
- https://github.com/department-of-veterans-affairs/va.gov-team/issues/79406
- https://github.com/department-of-veterans-affairs/va.gov-team/issues/81205

## Testing done

- [x] *New code is covered by unit tests*
Old behavior does not include a boolean `encrypted` state parameter in the logs.
This was manually tested as well as unit tested using respec.

## What areas of the site does it impact?
This impacts the evidence file upload for Decision Reviews related forms.

## Acceptance criteria

- [x]  I fixed|updated|added unit tests and integration tests for each feature (if applicable).
- [x]  No error nor warning in the console.
- [x]  Events are being sent to the appropriate logging solution
- [x]  No sensitive information (i.e. PII/credentials/internal URLs/etc.) is captured in logging, hardcoded, or specs

## Requested Feedback

(OPTIONAL)_What should the reviewers know in addition to the above. Is there anything specific you wish the reviewer to assist with. Do you have any concerns with this PR, why?_

### CODE REVIEW

File: app/controllers/v0/decision_review_evidences_controller.rb [hunk 1]
```diff
@@ -22,7 +22,8 @@ def save_attachment_to_cloud!
         downstream_system: 'AWS S3',
         params: {
           # `form_attachment` is declared in `FormAttachmentCreate`, included above.
-          form_attachment_guid: form_attachment&.guid
+          form_attachment_guid: form_attachment&.guid,
+          encrypted: !filtered_params[:password].nil?
```
Comment (anniebtran) [inline app/controllers/v0/decision_review_evidences_controller.rb:26]: i wonder if we might be able to flip this and do `filtered_params[:password].present?` to help with readability?

Comment (dfong-adh) [reply app/controllers/v0/decision_review_evidences_controller.rb:26]: Sure, will do 👍

**Review by anniebtran — DISMISSED** [review]
LGTM

**Comment by RachalCassity** [comment]
@dfong-adh @anniebtran To ensure CODEOWNERS are accurate. Are the CODEOWNERS currently assigned to` app/controllers/v0/decision_review_evidences_controller.rb `and `spec/controllers/v0/decision_review_evidences_controller_spec.rb` correct? Does another team need to be added as a CODEOWNER?

Current CODEOWNERS:
`app/controllers/v0/decision_review_evidences_controller.rb @department-of-veterans-affairs/Benefits-Team-1 @department-of-veterans-affairs/va-api-engineers @department-of-veterans-affairs/backend-review-group`

`spec/requests/decision_review_evidences_request_spec.rb @department-of-veterans-affairs/Benefits-Team-1 @department-of-veterans-affairs/va-api-engineers @department-of-veterans-affairs/backend-review-group`

**Comment by anniebtran** [comment]
@RachalCassity ooh good catch! I think we need to swap out `.../Benefits-Team-1` for `@department-of-veterans-affairs/benefits-decision-reviews-be`

Per this [ticket](https://github.com/department-of-veterans-affairs/va.gov-team/issues/66191) — seems like we're phasing out `Benefits-Team-1` in favor of more specific teams. cc: @shaunburdick

**Review by RachalCassity — APPROVED** [review]

**Comment by RachalCassity** [comment]
@anniebtran Perfect! This will also allow your team to approve code changes and not rely on the backend. Please feel free to replace other references to Benefits-Team-1 .

**Pull request merged**